crates/driver/src/infra/config/file/load.rs

```diff
@@ -40,9 +41,12 @@ pub async fn load(chain: chain::Id, path: &Path) -> infra::Config {
     });
 
     assert_eq!(
-        config.chain_id.map(chain::Id::from).unwrap_or(chain),
+        config
+            .chain_id
+            .and_then(|id| Chain::try_from(id).ok())
```

Comment:
Using `ok()` here discards `ChainIdNotSupported` error so message log would be misleading that chain id doesn't match connected node instead of not supported chain id.

Reply:
You are right! Adjusted.